### PR TITLE
Configurable JSON parser limit

### PR DESCRIPTION
**Pull request by gabriel-indik**
Signed-off-by: Gabriel Indik <gabriel.indik@kaleido.io>

### CODE REVIEW

**Review by eberger727 — APPROVED** [review]

File: src/lib/utils.ts [hunk 1]
```diff
@@ -44,7 +44,8 @@ export const constants = {
   MAX_EVENT_QUEUE_SIZE: 1000,
   HASH_HEADER_NAME: 'dx-hash',
   SIZE_HEADER_NAME: 'dx-size',
-  LAST_UPDATE_HEADER_NAME: 'dx-last-update'
+  LAST_UPDATE_HEADER_NAME: 'dx-last-update',
+  JSON_PARSER_LIMIT: process.env.JSON_PARSER_LIMIT || '1mb'
```
Comment (peterbroadhurst) [inline src/lib/utils.ts:48]: I was looking and today the only env var based configuration is `LOG_LEVEL` and `DATA_DIRECTORY`, and the rest of the configuration is in `config.json`.

Would it make more sense for this to be a `api.jsonBodyLimit` config.json?

If not, maybe a guide (in comments in the code?) would make sense for contributors to know when to add a config parameter as a `config.json` vs. an env var.

Comment (gabriel-indik) [reply src/lib/utils.ts:48]: @peterbroadhurst will be updating this PR to surface the payload size limit in the configuration file, I'll also create a new FireFly issue to ensure that when pinned private messages are batched, their combined payload does not go beyond a configurable limit. FireFly would also need to outright reject single private messages whose payload size go beyond that configurable limit.

**Comment by gabriel-indik** [comment]
A new optional property has been added to the configuration file. The property name is `jsonParserLimit` its value is a string that follows the pattern specified in the schema: `^\\d+(k|K|m|M|g|G|t|T|p|P)?(b|B)$`. Examples of valid values: `50b`, `50kb`, `50mb`. Since the property is optional, this is not a breaking change. The default value that is applied when the property is not present in the configuration file is `1mb`.

**Comment by gabriel-indik** [comment]
Opened https://github.com/hyperledger/firefly/issues/386 to track the work to add a configurable limit on the FireFly side.

**Review by peterbroadhurst — APPROVED** [review]

**Pull request merged**